plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -65,3 +67,30 @@ inline fun <reified I> VerificationError.getInfoOrNull(): I? =
         Info.fromSilver(result.reason().offendingNode().info).unwrapOr<I> { null }
     }
 
+/**
+ * Extracts information from a specific argument of a function involved in a `VerificationError`.
+ * This function is specifically designed to work with errors related to preconditions
+ * in function calls (`PreconditionInCallFalse`) or failed assertions (`AssertFailed`).
+ */
+fun VerificationError.extractInfoFromFunctionArgument(index: Int): Info {
```

Comment:
I'm a bit confused about why all this is necessary.  Why not just get the reason and use the function below, maybe throwing in case it isn't a `FuncApp`?  Or, more generally, why not add a `child` function that just gets the relevant child?  It seems like we are going to have more cases when we need to deal with extracting info from nested Viper structures.

Reply:
The function is defined on `VerificationError` because of the following line here:
```kotlin
error is PreconditionInCallFalse || error is AssertFailed)
```
---
> Why not just get the reason and use the function below, maybe throwing in case it isn't a FuncApp? 

Sadly, I cannot do this due to the following error:

```
Cannot access 'viper.silver.verifier.AbstractError' which is a supertype of 'viper.silver.verifier.VerificationError'. Check your module classpath for missing or conflicting dependencies
```

I cannot reference Viper's structures from the `VerifierErrorInterpreter` object.

> why not add a child function that just gets the relevant child? 

I think I cannot do that for the same reason as above.

Reply:
But `VerificationError` is our own interface.  If you want access to `reason` from it, you can add it there, I think?

Reply:
True, but then I think I should wrap the error type of `reason`, similar to what we do with `Position`, `Info` and so on.

Reply:
Isn't the reason basically just the offending node?  You could get the node directly, I suspect.

Reply:
That's true, but then I think Viper's type will be exposed again, since our custom classes do not wrap it.

Reply:
We can work on a translation from Silver types back to our wrappers if necessary, but I would keep it simple for now.